src/web/js/error-ui.js

```diff
@@ -55,6 +55,46 @@
       }
 
       // HELPERS
+      // NOTE: MUST BE CALLED WHEN RUNNING ON runtime's STACK
+      function renderSimpleReason(err) {
+        var ret = Q.defer();
+        runtime.runThunk(function() { 
+          return get(err.exn || err, "render-reason").app(); 
```

Comment:
Oof.  This `err.exn || err` is a real annoyance.  I know it's because there are inconsistent representations coming from somewhere else.  I wonder if a single helper that takes an "error-shaped" thing and produces a Pyret object for it would be useful.  I'm worried about these short little hacks littering the code and leading to accidental undefined values.